photonlib-java-examples/build.gradle

```diff
@@ -1,35 +1,145 @@
+import edu.wpi.first.toolchain.NativePlatforms
+
+plugins {
+    id 'edu.wpi.first.NativeUtils' version '2021.1.1'
+}
+
 apply plugin: 'java'
+apply plugin: 'cpp'
+apply plugin: 'edu.wpi.first.NativeUtils'
 
-String wpilibVersion = "2021.1.2"
-String opencvVersion = "3.4.7-5"
-String ejmlVersion = "0.38"
-String jacksonVersion = "2.10.0"
+// This adds all of the WPILib dependencies.
+nativeUtils.addWpiNativeUtils()
+nativeUtils {
+    wpi {
+        configureDependencies {
+            wpiVersion = "2021.1.2"
+            opencvVersion = "3.4.7-5"
+            wpimathVersion = "2021.1.2"
+        }
+    }
+    dependencyConfigs {
+        gui {
+            groupId = "edu.wpi.first.halsim"
+            artifactId = "halsim_gui"
+            headerClassifier = "headers"
+            sourceClassifier = "sources"
+            ext = "zip"
+            version = "2021.1.2"
+            sharedPlatforms << "osxx86-64" << "linuxx86-64" << "windowsx86-64"
+        }
+    }
+    combinedDependencyConfigs {
+        halsim_gui {
+            libraryName = "halsim_gui"
+            targetPlatforms << "osxx86-64" << "linuxx86-64" << "windowsx86-64"
+            dependencies << "gui_shared"
+        }
+    }
+}
 
 repositories {
     maven {
         url = 'https://frcmaven.wpi.edu:443/artifactory/release'
     }
 }
 
+ext {
+    exampleFile = new File("$projectDir/src/main/java/org/photonlib/examples/examples.json")
+}
+
 dependencies {
     implementation project(':photon-lib')
     implementation project(':photon-targeting')
 
-    implementation "edu.wpi.first.wpilibj:wpilibj-java:${wpilibVersion}"
-    implementation "edu.wpi.first.wpimath:wpimath-java:${wpilibVersion}"
-    implementation "edu.wpi.first.ntcore:ntcore-java:${wpilibVersion}"
-    implementation "edu.wpi.first.wpiutil:wpiutil-java:${wpilibVersion}"
-    implementation "edu.wpi.first.thirdparty.frc2021.opencv:opencv-java:${opencvVersion}"
-    implementation "edu.wpi.first.cscore:cscore-java:${wpilibVersion}"
-    implementation "edu.wpi.first.cameraserver:cameraserver-java:${wpilibVersion}"
-    implementation "edu.wpi.first.hal:hal-java:${wpilibVersion}"
-    implementation "org.ejml:ejml-simple:${ejmlVersion}"
-    implementation "com.fasterxml.jackson.core:jackson-annotations:${jacksonVersion}"
-    implementation "com.fasterxml.jackson.core:jackson-core:${jacksonVersion}"
-    implementation "com.fasterxml.jackson.core:jackson-databind:${jacksonVersion}"
+    implementation 'edu.wpi.first.cscore:cscore-java:2021.+'
```

Comment:
This may need to be reconciled with the other build.gradle's to ensure the strategy for picking wpilib version matches - I assume this picks "2021 latest", as suppsed to a hardcoded version?

Reply:
yeah it'd pick up the latest 2021 build. And I think that includes dev releases, if we have the dev maven repo added?